charms/istio-pilot/src/charm.py

```diff
@@ -401,6 +412,39 @@ def _get_ingress_auth_data(self) -> dict:
 
         return ingress_auth_data
 
+    def _get_ingress_data(self, event) -> dict:
+        """Retrieve the ingress relation data without touching other interface data.
+
+        This is a workaround to ensure that errors in other relation data, such as an incomplete
+        ingress-auth relation, do not block us from retrieving the ingress data.
+        """
+        try:
+            ingress_interface = get_interface(self, INGRESS_RELATION_NAME)
+        except NoVersionsListed as err:
+            raise ErrorWithStatus(err, WaitingStatus)
```

Comment:
should not this be `str(err)` ?

Reply:
For convenience, ErrorWithStatus's `msg` arg is [always converted to a string](https://github.com/canonical/charmed-kubeflow-chisme/blob/592c8d8eba7d655f576c8ac8971451ffb31a4251/src/charmed_kubeflow_chisme/exceptions/_with_status.py#L16) in the constructor, so I think this is ok?